classes/monograph/PublishedMonographDAO.inc.php

```diff
@@ -299,6 +311,95 @@ function updateObject($publishedMonograph) {
 			)
 		);
 	}
+
+	/**
+	 * Map a column heading value to a database value for sorting
+	 * @param $sortBy string
+	 * @return string
+	 */
+	static function getSortMapping($sortBy) {
+		switch ($sortBy) {
+			case ORDERBY_TITLE:
+				return 'st.setting_value';
+			case ORDERBY_DATE_PUBLISHED:
+				return 'ps.date_published';
+			case ORDERBY_SERIES_POSITION:
+				return 's.series_position';
+			default: return null;
+		}
+	}
+
+	/**
+	 * Get possible sort options.
+	 * @return array
+	 */
+	function getSortSelectOptions() {
+		return array(
+			ORDERBY_TITLE . SORT_DIRECTION_ASC => __('catalog.sortBy.titleAsc'),
+			ORDERBY_TITLE . SORT_DIRECTION_DESC => __('catalog.sortBy.titleDesc'),
+			ORDERBY_DATE_PUBLISHED . SORT_DIRECTION_ASC => __('catalog.sortBy.datePublishedAsc'),
+			ORDERBY_DATE_PUBLISHED . SORT_DIRECTION_DESC => __('catalog.sortBy.datePublishedDesc'),
+			ORDERBY_SERIES_POSITION . SORT_DIRECTION_ASC => __('catalog.sortBy.seriesPositionAsc'),
+			ORDERBY_SERIES_POSITION . SORT_DIRECTION_DESC => __('catalog.sortBy.seriesPositionDesc'),
+		);
+	}
+
+	/**
+	 * Get sort option.
+	 * @param $sortBy string
+	 * @param $sortDir int
+	 * @return string
+	 */
+	function getSortOption($sortBy, $sortDir) {
+		return $sortBy . $sortDir;
```

Comment:
Maybe include a separator? That might make it possible to explode the string by the separator in other areas, to avoid having to comprehensively list all options.